plugins/s3/builtins/plugins.cpp

```diff
@@ -0,0 +1,84 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/tql2/plugin.hpp>
+
+#include "operator.hpp"
+
+namespace tenzir::plugins::s3 {
+
+template <template <class, detail::string_literal = ""> class Adapter,
+          class Plugin>
+class plugin2 final : public virtual operator_plugin2<Adapter<Plugin>> {
+public:
+  auto initialize(const record& plugin_config,
```

Comment:
@dominiklohmann The fact that we now need separate plugins for the operators also means that there now is a requirement for `load_s3.yaml` and `save_s3.yaml`, which is a non-trivial change to TQL1 and may also be confusing for users.

@raxyte suggested a `config_name()` function (which then just defaults to `name()`), to override the config filename, similar to how we have `operator_name()`.

Reply:
I'd be fine with that.

Reply:
We decided to not change the plugin API and instead just have `load_s3` and `save_s3` initialize from the global config.